### ISSUE TITLE
Refactor curves

### ISSUE THREAD

**Issue by kwinkunks** [agile-geoscience/welly#103]
Hate to say it, but I think curves need to become either Pandas `Series` or  `xarray` datasets (which are implemented via `pandas`). Essentially, all my `basis` stuff is just trying to reimplement indexing, and that what `pandas` does. 

However, this isn't quite as simple as just making a `DataFrame`. I think we must continue to allow each curve to have its own index (i.e. be on a different sample interval, or have a different range).

**Comment by fahdi104** [comment]
ah finally I found what I need. I'm looking for info what is basis actually is. Based on your comments, apparently its just some indexing. I am also trying to find if we can put its own index (depth) to any arbitrary curves.

**Comment by kwinkunks** [comment]
Right, it's indexing. Just implemented in a wonky way, compared to `pandas`. Each curve has its own. It is computed from `start` and `step`, and has the same length as the curve. (That is, it is not stored as an array but computed on the fly.)
